### PR TITLE
Fix TestNetworkPlugins/group/auto/HairPin 

### PR DESCRIPTION
**Pull request by ilya-zuyev**
Fixes https://github.com/kubernetes/minikube/issues/11285

We started to use kindnet cni by default for containerd and cri-o, hairpinMode it always true for it, so let's just fix the test

### CODE REVIEW

File: test/integration/net_test.go [hunk 1]
```diff
@@ -195,6 +183,23 @@ func TestNetworkPlugins(t *testing.T) {
 	})
 }
 
+// verifyHairpinMode makes sure the hairpinning (https://en.wikipedia.org/wiki/Hairpinning) is correctly configured for given CNI
+func verifyHairpinMode(t *testing.T, ctx context.Context, profile string, hairpin bool) {
```
Comment (medyagh) [inline test/integration/net_test.go:187]: why this didnt get added to the Integration Test Docs ?
https://minikube.sigs.k8s.io/docs/contrib/tests/

maybbe the `make generate-docs` expects it to be called validateHairPinMode...

File: test/integration/net_test.go [hunk 2]
```diff
@@ -195,6 +183,23 @@ func TestNetworkPlugins(t *testing.T) {
 	})
 }
 
+// verifyHairpinMode makes sure the hairpinning (https://en.wikipedia.org/wiki/Hairpinning) is correctly configured for given CNI
```
Comment (medyagh) [inline test/integration/net_test.go:186]: add more details, what exactly are we testing ? consider wrtiting this comment for someone who didnt know what is hairpin before

**Review by medyagh — CHANGES_REQUESTED** [review]

**Comment by medyagh** [comment]
fix lint

File: test/integration/net_test.go (hunk 1 above)
Comment (ilya-zuyev) [reply test/integration/net_test.go]: yeah, `generate-docs` looks only for `validate*` functions. fixed

**Comment by medyagh** [comment]
@ilya-zuyev please run `make generate-doc`

**Review by medyagh — APPROVED** [review]

**Pull request merged**